framework/src/utils/lib/data-vpc-props.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import { RemovalPolicy } from 'aws-cdk-lib';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { RetentionDays } from 'aws-cdk-lib/aws-logs';
+
+/**
+ * The properties for the DataVpc construct
+ */
+
+export interface DataVpcProps {
+  /**
+   * The CIDR to use to create the subnets in the VPC.
+   */
+  readonly vpcCidr: string;
+  /**
+   * The KMS key for the VPC flow log group
+   * @default - A new KMS key is created
+   */
+  readonly flowLogKey?: IKey;
+  /**
+   * The IAM role for the VPC flow log
+   * @default - A new IAM role is created
+   */
+  readonly flowLogRole?: IRole;
```

Comment:
can you please update to specify which principal should be supplied to the role for it to be assumed correctly by AWS?

https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-cwl.html#flow-logs-iam-role